### PR TITLE
Remove Xcode 12 constraint for `datatransport / watchos-test` job

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
This PR unblocks CI in #81.

So in #81, the `datatransport / watchos-test` job failed for reasons unrelated to the PR. The issue was that the job was using Xcode 12, and [recent changes to FirebaseCoreInternal](https://github.com/firebase/firebase-ios-sdk/pull/10075) caused [failures](https://github.com/google/GoogleDataTransport/runs/7707506918?check_suite_focus=true) since the changes aren't backwards compatible to Xcode 12 (which is expected).

So, this PR was opened to remove that Xcode constraint. But, in doing so, the `--destination` when building for watchOS (as defined by the scripts/build.sh in the firebase-ios-sdk) caused failures as it was set to iPhone 11 Pro which no longer exists on the latest Xcode. So, after a series of PRs (https://github.com/firebase/firebase-ios-sdk/pull/10078, https://github.com/firebase/firebase-ios-sdk/pull/10079, https://github.com/firebase/firebase-ios-sdk/pull/10085, https://github.com/firebase/firebase-ios-sdk/pull/10082, & #83), this PR was finally able to successfully run CI.

All of this was a good exercise to get the GHA infra of GDT in-sync with that of Firebase (#83, #84).

#no-changelog

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
Fine to delete this.

**Comment by ncooke3** [comment]
@paulb777, the problem is actually in [here](https://github.com/firebase/firebase-ios-sdk/blob/7ea72745e84b67f47bfa8a3d08c41ddc914de56a/scripts/build.sh#L169) in firebase-ios-sdk's `scripts/build.sh`.

So using the `scripts/build.sh` with watchOS will attempt to use a simulator from Xcode 12. I assume it is ok to send a PR to update that.

**Comment by ncooke3** [comment]
This should now be unblocked via the merging of https://github.com/firebase/firebase-ios-sdk/pull/10082

Manually kicking off each failed workflow

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,8 @@
 # commit Gemfile and Gemfile.lock.
 source 'https://rubygems.org'
 
-gem 'cocoapods', '1.10.1'
+gem 'cocoapods', '1.11.3'
 
 gem 'cocoapods-generate', '2.0.1'
-gem 'danger', '6.1.0'
+gem 'danger', '8.4.5'
+gem 'xcpretty'
```
Comment (ncooke3) [inline Gemfile:9]: I updated the gems to match that of the Gemfile in the Firebase repo.

I also added xcpretty as a means to fix the `error: xcpretty command not found` issue. Based on my learnings with the .ruby-version, I suspect that if this is to work, the Gemfile needs to be checked in. I can test and undo this change to the Gemfile/Gemfile.lock if that is not the case. For now, to get this PR's CI green, I have explicitly installed `xcpretty` in the `./setup-scripts.sh` file.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**